src/backend/distributed/transaction/transaction_management.c

```diff
@@ -628,16 +628,25 @@ CoordinatedSubTransactionCallback(SubXactEvent event, SubTransactionId subId,
 		 */
 		case SUBXACT_EVENT_START_SUB:
 		{
+			MemoryContext previousContext = CurrentMemoryContext;
+			MemoryContextSwitchTo(CitusXactCallbackContext);
+
 			PushSubXact(subId);
 			if (InCoordinatedTransaction())
 			{
 				CoordinatedRemoteTransactionsSavepointBegin(subId);
```

Comment:
we should avoid doing allocations in these functions (e.g. use sprintf on a char array on the stack), since they're currently leaking into the memory context

Reply:
Pushed some changes for this but following functions still use StringInfo's, is that okay ?:

// because this acts on variable length strings
StartRemoteTransactionBegin

// Because those are supposed to return the string buffers they create
//
// Could pass some stack allocated buffers into those functions but not
// sure if that would worth ?
BeginAndSetDistributedTransactionIdCommand
AssignDistributedTransactionIdCommand

Reply:
it's ok for them to allocate if they don't allocate in CitusXactCallbackContext

Reply:
None of those three seem to be allocating in CitusXactCallbackContext, so I agree that it's okay to skip them.